app/src/main/java/com/github/lookupgroup27/lookup/model/map/Renderer.kt

```diff
@@ -92,31 +99,54 @@ class Renderer : GLSurfaceView.Renderer {
 
     // Initialize the SkyBox
     skyBox = SkyBox()
+    textureManager = TextureManager(context) // Initialize texture manager
+    skyBoxTextureHandle =
+        textureManager.loadTexture(R.drawable.skybox_texture) // Load skybox texture
+    skyBox = SkyBox() // Initialize the skybox
+    intializeObjects() // Initialize other renderable objects
   }
 
+  /**
+   * Called to redraw the frame. Clears the screen, updates the camera view, and renders objects in
+   * the scene.
+   *
+   * @param unused the GL10 interface, not used
+   */
   override fun onDrawFrame(unused: GL10) {
+    GLES20.glClear(GLES20.GL_COLOR_BUFFER_BIT or GLES20.GL_DEPTH_BUFFER_BIT) // Clear the screen
 
-    // Clear the screen
-    GLES20.glClear(GLES20.GL_COLOR_BUFFER_BIT or GLES20.GL_DEPTH_BUFFER_BIT)
-
-    GLES20.glDepthMask(false) // Disable depth writing
+    GLES20.glDepthMask(false) // Disable depth writing for the skybox
 
-    // Bind the texture and render the SkyBox
-    textureManager.bindTexture(skyBoxTextureHandle)
+    textureManager.bindTexture(skyBoxTextureHandle) // Bind skybox texture
+    skyBox.draw(camera) // Render skybox
 
-    // Use this MVP matrix to render the skybox
-    skyBox.draw(camera)
+    GLES20.glDepthMask(true) // Re-enable depth writing
 
-    GLES20.glDepthMask(true) // Re-enable depth writing for other objects
+    drawObjects() // Render other objects
   }
 
+  /**
+   * Called when the surface dimensions change. Adjusts the viewport and updates the projection
+   * matrix based on the new aspect ratio.
+   *
+   * @param unused the GL10 interface, not used
+   * @param width the new width of the surface
+   * @param height the new height of the surface
+   */
   override fun onSurfaceChanged(unused: GL10, width: Int, height: Int) {
-    // Specify the size of the rendering window
-    GLES20.glViewport(0, 0, width, height)
+    GLES20.glViewport(0, 0, width, height) // Set viewport dimensions
+    val ratio: Float = width.toFloat() / height.toFloat() // Calculate aspect ratio
+    camera.updateProjectionMatrix(ratio) // Update camera projection matrix
+  }
 
-    val ratio: Float = width.toFloat() / height.toFloat()
+  /** Initializes additional objects in the scene. Currently includes a planet. */
+  private fun intializeObjects() {
+    planet = Planet(context, textureId = R.drawable.planet_texture) // Create planet
+  }
 
-    camera.updateProjectionMatrix(ratio)
+  /** Renders additional objects in the scene using the current MVP matrix. */
+  private fun drawObjects() {
+    planet.draw(camera) // Render the planet
   }
```

Comment:
Does this method add significant value? It seems like it might not be necessary. What do you think?

Reply:
I think it's a method that will help structure the code better once we will implement all types of objects (stars, constellations, etc.)